chain-signatures/node/src/rpc.rs

```diff
@@ -674,3 +732,154 @@ async fn try_publish_eth(
+                .inc();
+        })?
+        .watch()
+        .await;
+
+    //let chain = Chain::Ethereum;
+    match result {
+        Ok(tx_hash) => {
+            tracing::info!(
+                tx_hash = ?tx_hash,
+                num_requests = num_requests,
+                "published ethereum signatures successfully"
+            );
+            crate::metrics::NUM_SIGN_SUCCESS
+                .with_label_values(&[chain.as_str(), near_account_id.as_str()])
+                .inc_by(num_requests as f64);
+            for action in actions {
+                crate::metrics::SIGN_TOTAL_LATENCY
+                    .with_label_values(&[chain.as_str(), near_account_id.as_str()])
+                    .observe(action.request.indexed.timestamp.elapsed().as_secs_f64());
+                if action.request.indexed.timestamp.elapsed().as_secs() <= 30 {
+                    crate::metrics::NUM_SIGN_SUCCESS_30S
+                        .with_label_values(&[chain.as_str(), near_account_id.as_str()])
+                        .inc();
+                }
+            }
+            crate::metrics::SIGN_RESPOND_LATENCY
+                .with_label_values(&[chain.as_str(), near_account_id.as_str()])
+                .observe(start.elapsed().as_secs_f64());
+
+            Ok(())
+        }
+        Err(err) => {
+            tracing::error!(
+                error = ?err,
+                "failed to publish ethereum signature"
+            );
+            crate::metrics::SIGNATURE_PUBLISH_FAILURES
+                .with_label_values(&[chain.as_str(), near_account_id.as_str()])
+                .inc_by(num_requests as f64);
+            Err(())
+        }
+    }
+}
+
+async fn execute_batch_publish(
+    client: &ChainClient,
+    actions: &mut Vec<PublishAction>,
+    near_account_id: &AccountId,
+    start: Instant,
+) {
+    let mut signatures: HashMap<SignId, Signature> = HashMap::new();
+
+    for action in actions.iter() {
+        let expected_public_key =
+            mpc_crypto::derive_key(action.public_key, action.request.indexed.args.epsilon);
+
+        // We do this here, rather than on the client side, so we can use the ecrecover system function on NEAR to validate our signature
+        let Ok(signature) = crate::kdf::into_eth_sig(
+            &expected_public_key,
+            &action.output.big_r,
+            &action.output.s,
+            action.request.indexed.args.payload,
+        ) else {
+            tracing::error!(
+                sign_id = ?action.request.indexed.id,
+                "failed to generate a recovery id; trashing publish request",
+            );
+            return;
+        };
+        signatures.insert(action.request.indexed.id.clone(), signature);
+    }
+
+    let mut retry_count = 0;
+    loop {
+        let publish = match client {
+            ChainClient::Near(_) => {
+                unimplemented!("near has no batch publish");
```

Comment:
unimplemented can lead to a panic and crash of the node as a result
Let's log an error and mention that this is not implemented please.